src/main/php/PHPMD/AbstractNode.php

```diff
@@ -37,30 +37,28 @@ abstract class AbstractNode
     /**
      * The collected metrics for this node.
      *
-     * @var array<string, mixed>
+     * @var array<string, float|int|null>
      */
-    private $metrics = null;
+    private ?array $metrics = null;
 
     /**
      * Constructs a new PHPMD node.
      *
      * @param TNode $node
      */
     public function __construct(
-        private PDependNode $node,
+        private readonly PDependNode $node,
     ) {
     }
 
     /**
      * The magic call method is used to pipe requests from rules direct
      * to the underlying PDepend AST node.
      *
```

Comment:
Need the secound phpdoc to still specify the elements of the array:
```suggestion
     *
     * @param array<mixed> $args
```